crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -217,6 +217,12 @@ struct TypeInferenceBuilder<'db> {
 }
 
 impl<'db> TypeInferenceBuilder<'db> {
+    /// How big a string do we build before bailing?
+    ///
+    /// This is a fairly arbitrary number. It should be *far* more than enough
+    /// for most use cases, but we can reevaluate it later if useful.
+    const MAX_STRING_LITERAL_SIZE: usize = 4096;
```

Comment:
Nit: for this name to be accurate, we should also enforce it in `infer_string_literal_expression`, right?